src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_StorageFile3

```diff
@@ -0,0 +1,104 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Resources;
+using Windows.ApplicationModel.Resources.Core;
+
+// Given_StorageFile3 , as Given_StorageFile is used in PR#2407, and Given_StorageFile2 in PR#2449
+
+namespace Uno.UI.RuntimeTests.Tests
+{
+    [TestClass]
+    public class Given_StorageFile3
+    {
+        String _filename;
+        
+        [TestInitialize]
+        public void Init()
+        {
+            _filename = DateTime.Now.ToString("yyyyMMddHHmmss") + ".txt";
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+        }
+
+        [TestMethod]
+        public async void When_DateModified()
+        {
+            var folderForTestFile = Windows.Storage.ApplicationData.Current.LocalFolder;
+            Assert.IsNotNull(folderForTestFile, "cannot get LocalFolder - error outside tested method");
+
+            Windows.Storage.StorageFile testFile = null;
+            
+            try
+            {
+                testFile = await folderForTestFile.CreateFileAsync( _filename, Windows.Storage.CreationCollisionOption.FailIfExists);
+                Assert.IsNotNull(testFile, "cannot create file - error outside tested method");
+            }
+            catch
+            {
+                  Assert.Fail("CreateFile exception - error outside tested method");
+            }
+
+            DateTimeOffset dateBeforeModify = DateTimeOffset.Now;
+
+            // test of DateModified
+
+            // first, some wait - to be sure that returned date is not simply 'current date' or 'creation date'
+            // e.g. FAT has two seconds resolution - so wait should be longer
+            await Task.Delay(5000);
```

Comment:
Same as similar PRs, use a loop instead of an arbitrary wait delay.

Reply:
in "similar PR", I'm waiting for more specific info about this loop :)

Reply:
Use a loop with a small delay to check for your expected condition. Once reached, you can exist the test with success.

This way you're now forcing the test to take 5 seconds each time... only failed condition will reach this time.